From 6d1d4241ac9abc59970c4839845a2d174db58854 Mon Sep 17 00:00:00 2001
From: Frediano Ziglio <fziglio@redhat.com>
Date: Sat, 30 Jun 2018 05:46:19 +0100
Subject: [PATCH 38/43] Use proper invalid value for _vio_serial

For some reason kernel handles in Windows have 2 invalid values
depending on the type. Files/devices use INVALID_HANDLE_VALUE (-1),
while others use NULL (0). As _vio_serial is a file, created
with CreateFile, use INVALID_HANDLE_VALUE for consistency.

Signed-off-by: Frediano Ziglio <fziglio@redhat.com>
Acked-by: Christophe Fergeau <cfergeau@redhat.com>
---
 vdagent/vdagent.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vdagent/vdagent.cpp b/vdagent/vdagent.cpp
index 423c3ee..eed3103 100644
--- a/vdagent/vdagent.cpp
+++ b/vdagent/vdagent.cpp
@@ -198,7 +198,7 @@ VDAgent::VDAgent()
     , _session_is_locked (false)
     , _desktop_switch (false)
     , _display_setting (VD_AGENT_REGISTRY_KEY)
-    , _vio_serial (NULL)
+    , _vio_serial (INVALID_HANDLE_VALUE)
     , _read_pos (0)
     , _write_pos (0)
     , _logon_desktop (false)
-- 
2.17.1

